cmd/signer/main_test.go

```diff
@@ -0,0 +1,165 @@
+pdbAtsb1RTNAbMZsqQzM453DY8rhqarSLtXp2Zxq6vmEvJN+BEF8sHF8yUiAQq6T
+eGpyEb+oxy4wT+iPrlX0+o6Se0cfUvewa7+Ii8MV1f2I2kDvGVPvBJMhPi86FhmI
+M2pxLVOFAgMBAAECggEAbI58GaE3lUB5Cc0xHNySv8XjJlX+0S/KwH4Ts8loiqaO
+V/u/dWG/bHCwyzB9XvvZZWMbYEHHg+yroG8Rn0osY1l7s30kqvxt9CMZ9CyeoV1U
+bMx1qehR9jdD1lLlGnjrIxTL78TOQCGu0sl6KakUf8lF/zPQeOJoT2tqD8AkOBao
+XmOlnsEmCW/PJGM/e/oBsuxNq2uUAVOEiOxGhZqIN7HWfhTinKHX28FJ0pLCkuQ/
+D7Si5icXEfrD1pD81Ft9v35sSQrU3YZk2jR0z/mDBfi/MPp2orTA5ZfHes+hpnEc
+7yLS4wHNa+v760l3L7q1UGPKVycZv484zKmYFIQGHQKBgQDVZg9drlqBIgHXFE4x
+wYXBZfD8t2ll/3lCN6xdBi/u1HP0ZGHICZ2l9lhHwXeeCEblY2Al2xjSdtnLfK5c
+CnSvwu9UV8h4sB4kjn37PL0SohtZJlFZpt6vresoDpQQrMmnjhgr+ot9ODlH1bLF
+yDnCtVXGlGYwPYAn40IxrmB5lwKBgQDIIolqxU7WkZ7nARhR++G3NeVqV8sB7GWX
+LyPrnoKbb556BSM2Ourcu/3uQfWnMwrc5J53tVHZ6LjrB4pSPa546LooWv5vY9QD
+ETwQzUKNfTiGT4qwkUdBs08fU9qmy0TT9y2SKbpe76GAS4oPbpYbk7W78jJRiWAZ
+tZelUcSnQwKBgHJ/BwGRmdetQmV+7JF/rt9cbdd6JR/n2cywiFeFCVTQQsK+1UP5
+/M7eBPHDGQX+lONg1WaaTpAl2qd2ZyrVJVRkd/q9+r7eZ93fYjLZnOyRc7D6gS1j
+/hkubHyajdEAlFXFRKzcCdmOwBUN0JST4IHav4IDf2ykos1D/vEfCX5TAoGBAMYd
+ykKzx3OI+/BZmSWvXqXq6Iv5FLF2vqqGs9xPMaOFPzAzXcQVVuHkB1+QVAmL8bjx
+aB3AlKJOSp/++uKmxMxUNdQ1H6JNBFd0/Cz1xGgkCYyLuRNI/W0Af9bXP5/VoPDj
+w2zpeeD4/rruDGFya44pDsJa44zrnQJWTSQOacnZAoGAM9XjNyoue1g7mrMCFaal
+ff7KnzqzMn9QuzIP/C9ImuUB/z/FG7Rn98oK66RJuPu4/oix7AU8UDI5JrD7Evwr
+WdEI/ZhvdTMMrho9fruTFINBU0LaGEBN/xt4cv9crGcEvenoPE0WRoxhZBb2disg
+agNtVIy2S07B+Eixp/gqPW8=
+-----END PRIVATE KEY-----
+`
+var testPublicKey = `-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA6DzRGVzbUElC6iUgfzJb
+HEgNdyfiXn5133Uzah676igyoiW3GMCAF8tqkwPY4m9/sG45Thx2Qm3VKLirn9Y2
+h783E5oZOWEhOSMHoleq/xJiRqKIOG1RBuWrkfJwJL5GptygKEX0CTfcSsYI3EEz
+uyFFWkHXpFblTy2kXX2tHuUjdSIqRASG6Cq/Zs97WgAojFl28lFBO1qRJXLay55x
+k0ZlRkD8CzHg8NTBPpJaTJmtyOCC3RYP1G2z6KZQvDXJbjRYP+/trkt6VEmb3y09
++di3Gwn2HV2hEtpGVJzL/j1AoxKa8ks1CivWA9AOcVP+lQEfBk0EiQcatl3sj8AR
+GwIDAQAB
+-----END PUBLIC KEY-----
+`
+var testPublicKey2Fail = `-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAptSGG9mcncjSXEe97kbx
+Eo8qjpRLUkNwGLzPHcmspFupF1KIVgTX4fxGBNJ4Ycj0TrcJm2ho3xkxLXcbWJX0
+R3i1qHf3QqV8VSEq2fZNzKZRttonT2b4MXj0q3allZP9pZTpZuQFGKFOwb6DL3k7
+hoJJ8srLA+nLv5j/H+InFMzVU4GFcTLoPZbK/cfiHwK8710rJz+XmNVenKXWwLbG
+9UUzQGzGbKkMzOOdw2PK4amq0i7V6dmcaur5hLyTfgRBfLBxfMlIgEKuk3hqchG/
+qMcuME/oj65V9PqOkntHH1L3sGu/iIvDFdX9iNpA7xlT7wSTIT4vOhYZiDNqcS1T
+hQIDAQAB
+-----END PUBLIC KEY-----
+`
+
+const publicKeyExponent = 65537
+
+func TestSignatureOK(t *testing.T) {
+	publicKeys := resources.ReadPublicRsaKeysAsset(testPublicKey)
+	key := readPrivateKey([]byte(testPrivateKey))
+	s, err := createFileSignature(key, []byte(testContent))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	signature := []byte(s)
+
+	valid := signatures.IsSignatureValid([]byte(testContent), []byte(signature), publicKeys)
+	if !valid {
+		t.Error("Valid signature not recognized to be valid.")
+	}
+}
+
+func TestSignatureFail(t *testing.T) {
```

Comment:
This test currently fails with `Signature recognized as valid but it should not.`.

Reply:
solved. a signature that was changed by adding characters at the end is still recognised as being valid. it is strange for me. i changed the test by changing the third character of the signature.